### PR TITLE
DOC: add Wachiou's week5 Blog post

### PR DESCRIPTION
**Pull request by WassCodeur**
- Add my week5 blog post

### CODE REVIEW

File: docs/source/posts/2024/2024-07-06-week5-wachiou-bouraima.rst [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+   - Modified ``__init__.py`` files to support lazy loading where necessary. This ensures that modules are only loaded when they are accessed for the first time
+
+3. Added Type Stubs (``__init__.pyi``):
+
+   - Adding type stubs (``__init__.pyi``) provides type hints for lazy-loading modules, improving code readability and maintainability
+
+4. **Improved module organization:**
+
+   - Improved module organization and structure to effectively support lazy loading.
```
Comment (itellaetxe) [inline docs/source/posts/2024/2024-07-06-week5-wachiou-bouraima.rst:36]: How did you do this? I mean, what is the rationale behind improving the structure?

**Review by itellaetxe — COMMENTED** [review]
LGTM Wachiou. Only a small doubt, see below :)

File: docs/source/posts/2024/2024-07-06-week5-wachiou-bouraima.rst (hunk 1 above)
Comment (WassCodeur) [reply docs/source/posts/2024/2024-07-06-week5-wachiou-bouraima.rst:36]: Merci @itellaetxe for your feedback. 

I made a first implementation of lazy loading in which I imported modules and functions into __init__.pyi files like this:
``
from .actor import (
    Container,
    _color_fa,
    ...
)
``

then I restified and improved it like this

``from .actor import (
    Container as Container,
    _color_fa as _color_fa,
    ...
)
``
to comply with SPEC1 recommendations: https://scientific-python.org/specs/spec-0001/#type-checkers

**Review by skoudoro — APPROVED** [review]
LGTM, merging, thanks @WassCodeur

**Pull request merged**